### PR TITLE
Annotate errors in test/e2e/network/service.go

### PR DESCRIPTION
**Pull request by palash25**
Related to: #34059

Removes all occurences of the "Expect(err).NotTo(HaveOccurred())"
antipattern from the aforementioned test file.

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

#### Release Note
```
None
```

### CODE REVIEW

**Comment by MrHohn** [comment]
Thanks for the PR!
/ok-to-test

File: test/e2e/network/service.go [hunk 1]
```diff
@@ -1070,11 +1070,11 @@ var _ = SIGDescribe("Services", func() {
 
 		By("deleting service " + serviceName1 + " to release NodePort")
 		err = t.DeleteService(serviceName1)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).To(HaveOccurred(), "failed to create service in namespace: %s", ns)
```
Comment (MrHohn) [inline test/e2e/network/service.go]: failed to delete service

File: test/e2e/network/service.go [hunk 2]
```diff
@@ -1043,7 +1043,7 @@ var _ = SIGDescribe("Services", func() {
 		service := t.BuildServiceSpec()
 		service.Spec.Type = v1.ServiceTypeNodePort
 		result, err := t.CreateService(service)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).To(HaveOccurred(), "failed to create service in namespace: %s", ns)
```
Comment (MrHohn) [inline test/e2e/network/service.go]: Include service name here?

File: test/e2e/network/service.go (hunk 1 above)
Comment (MrHohn) [reply test/e2e/network/service.go]: Include service name here?

File: test/e2e/network/service.go [hunk 3]
```diff
@@ -1070,11 +1070,11 @@ var _ = SIGDescribe("Services", func() {
 
 		By("deleting service " + serviceName1 + " to release NodePort")
 		err = t.DeleteService(serviceName1)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).To(HaveOccurred(), "failed to create service in namespace: %s", ns)
 
 		By("creating service " + serviceName2 + " with no-longer-conflicting NodePort")
 		_, err = t.CreateService(service2)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).To(HaveOccurred(), "failed to create service in namespace: %s", ns)
```
Comment (MrHohn) [inline test/e2e/network/service.go]: Include service name here?

File: test/e2e/network/service.go [hunk 4]
```diff
@@ -2137,10 +2137,10 @@ func execAffinityTestForNonLBService(f *framework.Framework, cs clientset.Interf
 	defer func() {
 		framework.Logf("Cleaning up the exec pod")
 		err := cs.CoreV1().Pods(ns).Delete(execPodName, nil)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).To(HaveOccurred(), "failed to delete pod: %s in namespace: %s", execPodName, ns)
 	}()
 	execPod, err := cs.CoreV1().Pods(ns).Get(execPodName, metav1.GetOptions{})
-	Expect(err).NotTo(HaveOccurred())
+	Expect(err).To(HaveOccurred(), "failed to delete pod: %s in namespace: %s", execPodName, ns)
```
Comment (MrHohn) [inline test/e2e/network/service.go]: failed to fetch pod

**Comment by palash25** [comment]
@MrHohn I have made the requested changes, could you please review it again? 

Thanks :smile:

File: test/e2e/network/service.go [hunk 5]
```diff
@@ -1043,7 +1043,7 @@ var _ = SIGDescribe("Services", func() {
 		service := t.BuildServiceSpec()
 		service.Spec.Type = v1.ServiceTypeNodePort
 		result, err := t.CreateService(service)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to create service: %s in namespace: %s", ns, serviceName1)
```
Comment (MrHohn) [inline test/e2e/network/service.go]: Reverse the order of `ns` and `serviceName1`

File: test/e2e/network/service.go [hunk 6]
```diff
@@ -1070,11 +1070,11 @@ var _ = SIGDescribe("Services", func() {
 
 		By("deleting service " + serviceName1 + " to release NodePort")
 		err = t.DeleteService(serviceName1)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to delete service: %s in namespace: %s", ns, serviceName1)
```
Comment (MrHohn) [inline test/e2e/network/service.go]: Reverse the order of `ns` and `serviceName1`

File: test/e2e/network/service.go [hunk 7]
```diff
@@ -1070,11 +1070,11 @@ var _ = SIGDescribe("Services", func() {
...
 
 		By("creating service " + serviceName2 + " with no-longer-conflicting NodePort")
 		_, err = t.CreateService(service2)
-		Expect(err).NotTo(HaveOccurred())
+		Expect(err).NotTo(HaveOccurred(), "failed to create service: %s in namespace: %s", ns, serviceName1)
```
Comment (MrHohn) [inline test/e2e/network/service.go]: Reverse the order of `ns` and `serviceName1`

**Review by MrHohn — APPROVED** [review]
Thanks, LGTM with some nits

**Comment by MrHohn** [comment]
@palash25  Could you also update the release note to `NONE`?
/kind cleanup

**Pull request merged**